lib/rubocop/cop/lint/script_permission.rb

```diff
@@ -30,29 +30,32 @@ module Lint
       #
       #   puts 'hello, world'
       #
-      class ScriptPermission < Cop
+      class ScriptPermission < Base
+        extend AutoCorrector
+
         MSG = "Script file %<file>s doesn't have execute permission."
         SHEBANG = '#!'
 
-        def investigate(processed_source)
+        def on_new_investigation
           return if @options.key?(:stdin)
           return if Platform.windows?
           return unless processed_source.start_with?(SHEBANG)
           return if executable?(processed_source)
 
           comment = processed_source.comments[0]
           message = format_message_from(processed_source)
-          add_offense(comment, message: message)
-        end
 
-        def autocorrect(node)
-          lambda do |_corrector|
-            FileUtils.chmod('+x', node.loc.expression.source_buffer.name)
+          add_offense(comment, message: message) do
+            autcocorrect(comment)
           end
         end
 
         private
 
+        def autcocorrect(comment)
```

Comment:
Oh! Interesting! 😅
This is a problem, because now the block is called even if autocorrection won't happen...
So this would always do the `chmod`.
I had not considered this case 😆, and there's no test for that either I think.
If you add `if autocorrect_requested?` you should be pretty much ok.

Reply:
> (Also, there's a typo)

Oops! I missed this typo 💦 

> This is a problem, because now the block is called even if autocorrection won't happen...

I did print debugging. The block execution seems to be running only when offensed. Is `if autocorrect_requested?` required?

Reply:
I added a spec in #8441
I think that if you rebase now, you will get a failure. Sorry, the `expect_offense` method need improvement...

Reply:
The build failed as expected with #8441. I added `if autocorrect_requested?` and the build has been passed. Thank you for adding the reproduction test.

Reply:
Great, thanks